tools/ci/actions/check-configure.sh

```diff
@@ -50,46 +50,37 @@ else
   COLOR='33'
 fi
 
-CI_SCRIPT='tools/ci/actions/check-configure.sh'
-PATHS=\
-'configure\|configure\.ac\|VERSION\|aclocal\.m4\|build-aux/.*'\
-'\|tools/autogen\|tools/git-dev-options\.sh'
-
 # $1 - commit to checkout files from
-# $2 - range of commits to diff
-# When testing a single commit, $1 and $2 will be the same; when validating the
-# tip of a branch, $1 will be HEAD and $2 will be the range of commits in the
-# branch.
 CheckTree () {
   RET=0
   COMMIT="$1"
-  COMMITS_TO_SEARCH="$2"
-  if git diff-tree --diff-filter=d --no-commit-id --name-only -r \
-       "$COMMITS_TO_SEARCH" | grep -qx "$PATHS\|$CI_SCRIPT"; then
-    git checkout -qB return
-    git checkout -q "$COMMIT"
-    mv configure configure.ref
-    tools/autogen
-    if diff -q configure configure.ref >/dev/null ; then
-      echo -e "$COMMIT: \e[32mconfigure.ac generates configure\e[0m"
-    else
-      RET=1
-      echo -e \
-        "$COMMIT: \e[${COLOR}mconfigure.ac doesn't generate configure\e[0m"
-    fi
-    mv configure.ref configure
-    git checkout -q return
+  git checkout -qB return
```

Comment:
The git manual suggests this equivalent:
```suggestion
  git branch -q --force return
```
The `return` here looks odd to me, it is the branch name? Maybe quoting it would make it clearer that it's not the shell keyword.

Reply:
Possibly, yes - the whole mechanism is a bit of a mess... just the usual nerves that modifying a part of the script which was working risks accidentally breaking it! 💣😊